web/api/formatters/rrd2json.h

```diff
@@ -53,12 +53,6 @@
 extern void rrd_stats_api_v1_chart(RRDSET *st, BUFFER *wb);
 extern void rrdr_buffer_print_format(BUFFER *wb, uint32_t format);
 
-typedef struct context_param {
```

Comment:
I've played a bit with the code and you don't really have to move anything to `rrd.h`.

The only issue is the `typedef` redefinition, which is not compliant with anything < C11. The minimum fix is to remove the `typedef` from the `struct` definition here, and keep `rrd.h` as is:
```
struct context_param {
Attach files by dragging & dropping, selecting or pasting them.
    RRDDIM *rd;
    time_t first_entry_t;
    time_t last_entry_t;
}; 
```

Reply:
Alternatively, if we should put the structure in `rrd.h`, then we should do it consistently as such:
-  keep `typedef struct context_param CONTEXT_PARAM;` in the forward typedefs.
- Add `struct context_param;` in the forward declarations but do not add the definition there.
- Skip ` #include <time.h>` as it will be included by ` #include "../daemon/common.h"`
- Add the definition in the main body of `rrd.h`:
```
struct context_param {
Attach files by dragging & dropping, selecting or pasting them.
    RRDDIM *rd;
    time_t first_entry_t;
    time_t last_entry_t;
}; 
```

Reply:
Yes as discussed 2nd option is better will do that